web/api/rest-api-design/Batching-Filtering-Pagination.md

```diff
@@ -0,0 +1,73 @@
+## Batching, Filtering, and Pagination
+
+### Filtering:
+
+- Topics covered: Elasticsearch and URL Parameters
+
+#### Elasticsearch:
+
+- Elasticsearch is a distributed RESTful search engine that is recommended for most filtering needs.
+- It works by sorting results by a relevance score, which is a way to quantify how well each result matches a query (higher score = more relevance, e.g. the more relevant the document being searched)
+- The score is also determined by whether the query is run via a query or filter context.
+- In a **query context** the following query clause is passed to a query parameter to calculate a relevance score: How well does this document match this query clause?
+- When a **filter context** is used, which happens mostly when filtering structured data, a query clause is passed to a filter parameter (e.g. `filter`). Some examples include:
+
+  - Does this timestamp fall into the range 2015 to 2016?
+  - Is the status field set to "published"?
+
+Examples of a query and filter context:
+
+```
+GET /_search
+{
+  "query": {
+    "bool": {
+      "must": [
+        { "match": { "title":   "Search" }},
+        { "match": { "content": "Elasticsearch" }}
+      ],
+      "filter": [
+        { "term":  { "status": "published" }},
+        { "range": { "publish_date": { "gte": "2015-01-01" }}}
```

Comment:
if your example is the range from 2015 to 2016, do you want to add an `"lt": "2016-01-01"` here?